docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/DockerBuildVariableContributor.java

```diff
@@ -31,9 +31,11 @@ public void buildVariablesFor(AbstractBuild build, Map<String, String> variables
             variables.put("JENKINS_CLOUD_ID", dockerComputer.getCloudId());
             try {
                 //replace http:// and https:// from docker-java to tcp://
-                final String dockerHost = new URIBuilder(dockerComputer.getCloud().serverUrl)
-                        .setScheme("tcp")
-                        .toString();
+                final URIBuilder uriBuilder = new URIBuilder(dockerComputer.getCloud().serverUrl);
+                if (uriBuilder.getScheme() != "unix") {
```

Comment:
In Java == is a reference comparison so depending on how the string was created you could have ```uriBuilder.getScheme() == "unix"``` be false even if ```uriBuilder.getScheme().equals("unix")``` is true.

Then if you code in both Scala and Java, you get to be really confused...